src/main/java/seedu/address/logic/parser/SeplendidParser.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.Messages.MESSAGE_UNKNOWN_COMMAND;
+
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.address.commons.core.SeplendidLogsCenter;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.HelpCommand;
+import seedu.address.logic.commands.LocalCourseAddCommand;
+import seedu.address.logic.commands.LocalCourseCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses user input into the SEPlendid CLI.
+ */
+public class SeplendidParser {
+
+    /**
+     * Used for initial separation of command, action word and arguments.
+     */
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile(
+            "(?<commandWord>\\S+)\\s(?<actionWord>\\S+)\\s(?<arguments>.*)");
+    private static final Logger logger = SeplendidLogsCenter.getLogger(SeplendidParser.class);
+
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param userInput full user input string.
+     * @return the command based on the user input.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public Command parseCommand(String userInput) throws ParseException {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
+        if (!matcher.matches()) {
+            // Note that we are using HelpCommand from ab3
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandWord = matcher.group("commandWord");
+        final String actionWord = matcher.group("actionWord");
+        final String arguments = matcher.group("arguments");
+
+        // Note to developers: Change the log level in config.json to enable lower level (i.e., FINE, FINER and lower)
+        // log messages such as the one below.
+        // Lower level log messages are used sparingly to minimize noise in the code.
+        logger.fine("Command word: " + commandWord + "; Action word: " + actionWord
+                + "; Arguments: " + arguments);
+
+        switch (commandWord) {
+
+        case LocalCourseCommand.COMMAND_WORD:
+            switch (actionWord) {
+            case LocalCourseAddCommand.ACTION_WORD:
```

Comment:
Is there any way to not do the nested switch case?

Reply:
Method extracted.